feat(Unity): Updated ANR detection explainer - Single vs Multi Threaded (#14834)
getsentry/sentry-docs: docs/platforms/unity/configuration/app-not-responding/index.mdx

## ANR Implementation Details

The ANR detection in the Unity SDK works differently depending on whether multithreading is available on the target platform:

### Platforms with Multi-Threading Support

> On platforms that support multithreading, the ANR detection works in two parts:

Reviewer: ```suggestion
On platforms that support multithreading, the ANR detection has two components:
```

1. A coroutine periodically resets a counter

> 2. A background thread increments and monitors this counter. It reports an ANR event when the threshold defined on `options.AnrTimeout` is exceeded

Reviewer: ```suggestion
2. A background thread increments and monitors this counter, triggering an ANR event when the value exceeds the threshold defined in `options.AnrTimeout`.
```

### Platforms without Multi-Threading Support - WebGL

On platforms without multi-threading support like WebGL, the ANR detection uses a slightly different approach:

> - A single coroutine using a `Stopwatch` to measure elapsed time

Reviewer: ```suggestion
- A single coroutine uses a `Stopwatch` to measure elapsed time
```

- When the coroutine detects that more time has passed than the ANR timeout, it reports the ANR event

> The drawback here is that the ANR detection relies on the game recovering from the not-responding state to report the event.

Reviewer: ```suggestion
The drawback here is that the ANR detection relies on the game recovering from the non-responsive state to report the event.
```
